src/lib/map.js

```diff
@@ -123,3 +123,12 @@ module.exports.getRoomScore = function (toRoom, fromRoom, opts = {}) {
   }
   return scores.WEIGHT_NEUTRAL ? scores.WEIGHT_NEUTRAL : PATH_WEIGHT_NEUTRAL
 }
+
+// Our own implementation of depricated function
+module.exports.isRoomAvailable = function (roomName) {
```

Comment:
Could you rename this function to `isRoomAvailableForCreeps` or something else that emphasizes that it doesn't apply to observers?